src/client/dfuse/dfuse.h

```diff
@@ -876,6 +873,9 @@ struct dfuse_inode_entry {
 
 	struct dfuse_cont        *ie_dfs;
 
+	/* Lock, used to protect readdir calls */
+	pthread_mutex_t           ie_lock;
```

Comment:
nit: if this is only to protect readdir calls, should readdir be in the name of the lock? Or use be extended in the future?

Reply:
I didn't spend long writing this patch ;). This lock should be removed in future for the reason stated above.